src/IssueViz.Security.UnitTests/Taint/TaintList/TaintIssuesControlViewModelTests.cs

```diff
@@ -439,14 +466,155 @@ public void HasServerIssues_IssuesChanged_RaisesPropertyChanged()
+
+            var eventHandler = new Mock<PropertyChangedEventHandler>();
+            testSubject.PropertyChanged += eventHandler.Object;
+
+            eventHandler.VerifyNoOtherCalls();
+
+            var selectedIssue = isSelectedNull ? null : Mock.Of<IAnalysisIssueVisualization>();
+
+            RaiseSelectionChangedEvent(selectionService, selectedIssue);
+
+            eventHandler.Verify(x => x(testSubject,
+                It.Is((PropertyChangedEventArgs args) =>
+                    args.PropertyName == nameof(testSubject.SelectedIssue))), Times.Once);
+
+            eventHandler.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        [Microsoft.VisualStudio.TestTools.UnitTesting.Description("Verify that there is no callback, selection service -> property -> selection service")]
+        public void SelectionChanged_IssueSelected_SelectionServiceNotCalledAgain()
+        {
+            var selectedIssue = Mock.Of<IAnalysisIssueVisualization>();
+            var storeIssues = new [] { selectedIssue };
+            var selectionService = new Mock<IIssueSelectionService>();
+
+            CreateTestSubject(storeIssues, selectionService: selectionService.Object);
+
+            RaiseSelectionChangedEvent(selectionService, selectedIssue);
+
+            selectionService.VerifySet(x => x.SelectedIssue = It.IsAny<IAnalysisIssueVisualization>(), Times.Never);
+        }
+
+        [TestMethod]
+        public void SetSelectedIssue_IssueSet()
+        {
+            var testSubject = CreateTestSubject();
+
+            var selection = new TaintIssueViewModel(Mock.Of<IAnalysisIssueVisualization>());
+            testSubject.SelectedIssue = selection;
+
+            testSubject.SelectedIssue.Should().Be(selection);
+        }
+
+        [TestMethod]
+        public void SetSelectedIssue_SelectionServiceIsCalled()
+        {
+            var selectionService = new Mock<IIssueSelectionService>();
+            var testSubject = CreateTestSubject(selectionService: selectionService.Object);
+
+            selectionService.Reset();
+
+            var selection = new TaintIssueViewModel(Mock.Of<IAnalysisIssueVisualization>());
+            testSubject.SelectedIssue = selection;
+
+            selectionService.VerifySet(x => x.SelectedIssue = selection.TaintIssueViz, Times.Once);
+            selectionService.VerifyNoOtherCalls();
+        }
+
+        [TestMethod]
+        public void SetSelectedIssue_ValueIsTheSame_SelectionServiceNotCalled()
+        {
+            var selectionService = new Mock<IIssueSelectionService>();
+            var testSubject = CreateTestSubject(selectionService: selectionService.Object);
+
+            var selection = new TaintIssueViewModel(Mock.Of<IAnalysisIssueVisualization>());
+            testSubject.SelectedIssue = selection;
+
+            selectionService.Reset();
+
+            testSubject.SelectedIssue = selection;
+
+            selectionService.VerifyNoOtherCalls();
+        }
```

Comment:
"One check per test" is a reasonable rule of thumb, but I find in practice there are scenarios it makes the tests harder to read, and it harder to see if all of the appropriate cases have been covered. 
This "selection service only called if selected object has changed" scenario is an example where I'd have considered testing multiple state transitions in the same test.